File: src/com/nerdscorner/mvp/mvp/busevents/activity/ActivityComponent.java
package com.nerdscorner.mvp.mvp.busevents.activity;

import java.io.File;

import com.nerdscorner.mvp.mvp.busevents.base.MvpComponent;

public class ActivityComponent extends MvpComponent {
    public static final String PACKAGE_SUFFIX = ".ui.activities";
    public static final String ACTIVITY_SUFFIX = "Activity";

    private static final String ACTIVITY_TEMPLATE = LANGUAGE_PLACEHOLDER + "/activity_template";
    private static final String ACTIVITY_JAVA = "Activity." + LANGUAGE_EXTENSION_PLACEHOLDER;

    private static final String RELATIVE_PATH = File.separator + "ui" + File.separator + "activities";

    public ActivityComponent(String basePath, String basePackage, String screenName, boolean shouldCreateWiring) {
        super(basePath, basePackage, screenName, ACTIVITY_TEMPLATE, screenName + ACTIVITY_JAVA, shouldCreateWiring);
    }

    @Override
    protected String relativePath() {
        return RELATIVE_PATH;
    }
}

File: src/com/nerdscorner/mvp/mvp/busevents/activity/ActivityComponent.kt
package com.nerdscorner.mvp.mvp.busevents.activity

import com.nerdscorner.mvp.mvp.busevents.base.MvpComponent
import java.io.File

class ActivityComponent(basePath: String, basePackage: String, screenName: String, shouldCreateWiring: Boolean)
    : MvpComponent(basePath, basePackage, screenName, ACTIVITY_TEMPLATE, screenName + ACTIVITY_JAVA, shouldCreateWiring) {

    override fun relativePath() = RELATIVE_PATH

    companion object {
        const val PACKAGE_SUFFIX = ".ui.activities"
        const val ACTIVITY_SUFFIX = "Activity"

        private const val ACTIVITY_TEMPLATE = "$LANGUAGE_PLACEHOLDER/activity_template"
        private const val ACTIVITY_JAVA = "Activity.$LANGUAGE_EXTENSION_PLACEHOLDER"

        private val RELATIVE_PATH = "${File.separator}ui${File.separator}activities"
    }
}
